hyperopt/fmin.py

```diff
@@ -237,9 +240,10 @@ def get_n_unfinished():
         ) as progress_ctx:
 
             all_trials_complete = False
-            while (n_queued < N or (block_until_done and not all_trials_complete)) and (
-                self.timeout is None or (timer() - self.start_time) < self.timeout
-            ):
+            best_loss = sys.maxsize
+            while ( ( n_queued < N or (block_until_done and not all_trials_complete) )
+              and   ( self.timeout is None or (timer() - self.start_time) < self.timeout )
+              and   ( self.loss_threshold is None or best_loss > self.loss_threshold) ):
```

Comment:
I think this set of conditions is getting tricky to read, can you add a comment above it to explain (not only your part but all of it)?

I would use >= rather than > for users who want to minimise something to exactly a number, e.g. 0

minor suggestion: use float("inf")  for initial value of best_loss?